### PR TITLE
Reduce support in view query

### LINKED ISSUE #12: View with Reducer
Hi, I have a code looking like: 

```
val view = db.query.view[String, Int]("myDoc", "myView").get
```

My view is defined with a reducer (`_sum`). When I run:

```
view.reduce(false).query.run
```

Everything is fine - no error.  On the other hand, when I run it with:

```
view.reduce(true).query.run
```

There is a pickling error: 

```
[info]   upickle.Invalid$Data: data: Obj(ArrayBuffer((rows,Arr(ArrayBuffer(Obj(ArrayBuffer((key,Null), (value,Num(16140.0))))))))) msg: Key Missing: offset
```

I was just wondering how I could get the output my reduced view? Thanks!

### PR DESCRIPTION
**Pull request by ermyas**
Fixes #12

### CODE REVIEW

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala [hunk 1]
```diff
@@ -112,14 +111,11 @@ case class GetManyDocumentsQueryBuilder(client: Client,
     queryByIds[CouchDocsIncludesMissing[String, CouchDocRev, D]](ids, includeDocs().params)
   }
 
-  private def queryWithoutIds[Q: R](parameters: Map[String, String]): Task[Q] = {
-    client.get[Q](s"/$db/_all_docs", Status.Ok, parameters.toSeq)
+  def queryWithoutIds[Q: R](ps: Map[String, String]): Task[Q] = {
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: Should this be private?

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala [hunk 2]
```diff
@@ -112,14 +111,11 @@ case class GetManyDocumentsQueryBuilder(client: Client,
...
+    query[Q](client, db, s"/$db/_all_docs", ps)
   }
 
-  private def queryByIds[Q: R](ids: Seq[String], parameters: Map[String, String]): Task[Q] = {
-    if (ids.isEmpty)
-      Res.Error("not_found", "No IDs specified").toTask[Q]
-    else
-      client.post[Req.DocKeys[String], Q](s"/$db/_all_docs", Status.Ok, Req.DocKeys(ids), parameters.toSeq)
+  def queryByIds[Q: R](ids: Seq[String], ps: Map[String, String]): Task[Q] = {
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: And this too?

File: src/main/scala/com/ibm/couchdb/api/builders/ViewQueryBuilder.scala [hunk 3]
```diff
@@ -116,40 +115,29 @@ case class ViewQueryBuilder[K, V](client: Client,
...
-    else
-      client.post[Req.DocKeys[K], CouchDocs[K, V, D]](
-        s"/$db/_design/$design/_view/$view",
-        Status.Ok,
-        Req.DocKeys(keys),
-        includeDocs().params.toSeq)
+    queryByIds[CouchDocs[K, V, D]](keys, includeDocs().params)
   }
 
+  def queryWithoutIds[Q: R](ps: Map[String, String]): Task[Q] = {
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/ViewQueryBuilder.scala]: private?

File: src/main/scala/com/ibm/couchdb/api/builders/ViewQueryBuilder.scala [hunk 4]
```diff
@@ -116,40 +115,29 @@ case class ViewQueryBuilder[K, V](client: Client,
...
+    query[Q](client, db, s"/$db/_design/$design/_view/$view", ps)
+  }
+
+  def queryByIds[Q: R](ids: Seq[K], ps: Map[String, String]): Task[Q] = {
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/ViewQueryBuilder.scala]: private?

**Comment by beloglazov** [comment]
I like how small a change was required to support reduce, nice!

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala (hunk 1 above)
Comment (ermyas) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: yes to all :-)

**Pull request merged**